lib/core/src/recover/recoverer.rs

```diff
@@ -171,6 +173,14 @@ impl Recoverer {
         &self,
         swaps: &mut [Swap],
     ) -> Result<HashMap<Txid, WalletTx>> {
+        let wallet_tip = self.onchain_wallet.tip().await;
+        let liquid_tip = self.liquid_chain_service.tip().await?;
+        if wallet_tip.abs_diff(liquid_tip) > LIQUID_TIP_LEEWAY {
```

Comment:
I am ok with this as a safety measure since it is better not to run the sync if the tips are not in sync but it seems that right now on restoring an old node the recovery will be skipped leaving the records without the swaps information.
Shouldn't we also ensure the sync is running after first_sync_completed?

Reply:
We only do a partial sync every 10 seconds (receive swaps), so in the negative flow we wait until the next block to do the full sync. Perhaps we should always full sync until the first sync is complete?

Reply:
I think if we do what Ross suggests then it is the only change necessary for this PR and we can end up with a tiny diff. I am in favor.

Reply:
@hydra-yse @dangeross  Going over this again and giving some more thinking I think the first sync is a private case. For example if the user opens a syncing wallet that was closed for a long time we can fall into the same issue.
What do you think about adding a wallet sync in the recovery only in the case we have a big diff (more than LIQUID_TIP_LEEWAY) in the blokchain tips?

Reply:
The LIQUID_TIP_LEEWAY is 3 blocks so I think even if I open a device after 24 hours of shutdown (which is totally reasonable) we will fall into this issue. That's why I suggested to run full sync in the recoverer in case we have a gap. I think now that we don't have performance/locking issues it should be fine.